demo/runtime/src/dispatch.rs

```diff
@@ -16,79 +16,230 @@
 
 //! Dispatch system. Just dispatches calls.
 
-use demo_primitives::{Function, Proposal, AccountId};
-use runtime::{staking, system, session, democracy, council, council_vote, timestamp};
-
-/// Dispatch a proposal.
-pub fn proposal(proposal: Proposal) {
-	match proposal {
-		Proposal::SystemSetCode(ref a) =>
-			system::privileged::set_code(a),
-		Proposal::SessionSetLength(a) =>
-			session::privileged::set_length(a),
-		Proposal::SessionForceNewSession =>
-			session::privileged::force_new_session(),
-		Proposal::StakingSetSessionsPerEra(a) =>
-			staking::privileged::set_sessions_per_era(a),
-		Proposal::StakingSetBondingDuration(a) =>
-			staking::privileged::set_bonding_duration(a),
-		Proposal::StakingSetValidatorCount(a) =>
-			staking::privileged::set_validator_count(a),
-		Proposal::StakingForceNewEra =>
-			staking::privileged::force_new_era(),
-		Proposal::DemocracyCancelReferendum(a) =>
-			democracy::privileged::cancel_referendum(a),
-		Proposal::DemocracyStartReferendum(a, b) =>
-			democracy::privileged::start_referendum(*a, b),
-		Proposal::CouncilSetDesiredSeats(a) =>
-			council::privileged::set_desired_seats(a),
-		Proposal::CouncilRemoveMember(a) =>
-			council::privileged::remove_member(&a),
-		Proposal::CouncilSetPresentationDuration(a) =>
-			council::privileged::set_presentation_duration(a),
-		Proposal::CouncilSetTermDuration(a) =>
-			council::privileged::set_term_duration(a),
-		Proposal::CouncilVoteSetCooloffPeriod(a) =>
-			council_vote::privileged::set_cooloff_period(a),
-		Proposal::CouncilVoteSetVotingPeriod(a) =>
-			council_vote::privileged::set_voting_period(a),
+use runtime::{staking, system};
+pub use rstd::prelude::Vec;
+pub use codec::{Slicable, Input, NonTrivialSlicable};
```

Comment:
re-exports in a random module worry me.

Reply:
it's for the macro.

Reply:
i understand that, but it's still bad style!